pkg/tsdb/prometheus/models/query.go

```diff
@@ -45,16 +45,12 @@ const (
 var safeResolution = 11000
 
 type QueryModel struct {
-	Expr           string `json:"expr"`
-	LegendFormat   string `json:"legendFormat"`
-	Interval       string `json:"interval"`
-	IntervalMS     int64  `json:"intervalMS"`
-	StepMode       string `json:"stepMode"`
-	RangeQuery     bool   `json:"range"`
-	InstantQuery   bool   `json:"instant"`
-	ExemplarQuery  bool   `json:"exemplar"`
-	IntervalFactor int64  `json:"intervalFactor"`
-	UtcOffsetSec   int64  `json:"utcOffsetSec"`
+	dataquery.PrometheusDataQuery
+	LegendFormat *string `json:"legendFormat,omitempty"`
```

Comment:
do we want "*string" -- rather than just string?  using "" as indication of missing (just like omitempty)

Reply:
@ryantxu I do not fully understand your point here. Can you elaborate on it, please?

Reply:
I believe he is asking if it can not be a pointer to a string. Can an empty string mean "not set"? If so probably doesn't need to be a pointer.